crypto/s2n_pkey.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "crypto/s2n_rsa.h"
+
+struct s2n_pkey {
+    union {
```

Comment:
If we eventually have multiple different key types in `s2n_pkey` (RSA, ECDSA, etc) how would we know which key in the union is actually populated and is safe to retrieve? It seems that `s2n_pkey` will always need to be accompanied by a `key_type` enum or similar so that users know which key in the union is populated.